### PR TITLE
feat: Add HyDE and reranking to evaluation

### PR DESCRIPTION
**Pull request by KevinJBoyer**
## Ticket

Resolves https://navalabs.atlassian.net/browse/DST-180

## Changes
 - Add HyDE and reranking to RAG prototype in eval pipeline

## Context for reviewers
 - For re-ranking, you can set up a free trial for Cohere and pass the COHERE_API_KEY environment variable
 - Preliminary results are available in the evaluation sheet here: https://docs.google.com/spreadsheets/d/1ZAZGZsqMGg2dX5qdy0j_n0xEAigWvffMPqvzvhD1DPk/edit#gid=821997362

### CODE REVIEW

File: 02-household-queries/eval.py [hunk 1]
```diff
@@ -30,6 +32,13 @@
 Please answer using the following context:
 {context}"""
 
+HYDE_PROMPT = """Please write a hypothetical document that would answer the following question about SNAP (food stamps.)
```
Comment (yoomlam) [inline 02-household-queries/eval.py:35]:
```suggestion
HYDE_PROMPT = """Please write a hypothetical document that would answer the following question about SNAP (food stamps).
```

File: 02-household-queries/requirements.in [hunk 2]
```diff
@@ -5,6 +5,7 @@
 beautifulsoup4
 chainlit
 chromadb
+cohere
```
Comment (yoomlam) [inline 02-household-queries/requirements.in:8]: Reminder to run `pip-compile requirements.in` to update `requirements.txt`

**Review by yoomlam — APPROVED** [review]
LGTM

**Comment by yoomlam** [comment]
I changed the base branch so that I only see the changes compared to PR #17, which I approved.

File: 02-household-queries/requirements.in (hunk 2 above)
Comment (KevinJBoyer) [reply 02-household-queries/requirements.in:8]: Read this comment, thought "ah right", and then pushed merge 😂  -- I'll update on the base branch now before merging.

**Pull request merged**